### ISSUE TITLE
watch : Exited with error code 139 watch : Waiting for a file to change before restarting dotnet...

### ISSUE THREAD

**Issue by Kwofie-mex** [dotnet/core#7112]
Problem encountered on https://dotnet.microsoft.com/en-us/learn/aspnet/blazor-tutorial/run
Operating System: linux

Provide details about the problem you're experiencing. Include your operating system version, exact error message, code sample, and anything else that is relevant.

**Comment by pranavkm** [comment]
139 usually means access violation. Is this error consistent? Do you get this running other `dotnet` commands?

**Comment by rosgori** [comment]
Yes! This is a little bit worrying. This happens when I want to run a small project in .NET 6 or .NET 5. This message is:

```
watch : Started

watch : Exited with error code 139
watch : Waiting for a file to change before restarting dotnet...
```

And that appears when I use the command `dotnet watch run` or `dotnet run`.  Also when I try to run from the folder `bin/Debug/My-proyecto`, another message appears: `Violación de segmento ('core' generado)`, which translates to English (if I am not mistaken) as `Core Dump (Segmentation fault)`. I don't know why this happens, but what is clear is I cannot use .NET in my machine.

Some info:

- Operating system: Ubuntu 20.04 with kernel `5.13.0-28-generic`.
- This happens with `dotnet` version 6.0.101 and the last from .NET 5 (I think it is 5.0.404).

**Comment by rosgori** [comment]
Same problem with `dotnet` version 6.0.102.

**Comment by danroth27** [comment]
Is it possible you're building from/to a protected directory?

**Comment by rosgori** [comment]
The solution is weird, I don't know why. I have installed `dotnet` through `snap` and it was working fine, no problem there. Suddenly on February when I tried to run some projects, it didn't work, it showed `watch : Exited with error code 139`. I tried two possible solutions: the first one, I downloaded the binaries from [here](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) and I used the executable (for the versions 5 and 6). Same error. The second solution was to install from apt (I am using Ubuntu) using this [guide](https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#2004-). **It worked!**. Again, I don't know why, but I can run any project using `dotnet run` or `dotnet watch run`. Although this does not answer the question: why did that kind of error appear?

**Comment by mot256** [comment]
We were debugging an issue where our .NET containers (linux) sometimes exit with 139 errors and I later found that if the .NET app (which runs as the entrypoint to the container) throws an unhandled exception (same as in @rosgori's [comment](https://github.com/dotnet/core/issues/7112#issuecomment-1031000223)), then the container consistently has an exit code of 139. 
What is even more confusing is that if you shell into the same container and run the "dotnet my-app.dll" manually you get an exit code of 134 (Abort) when an unhandled exception is thrown. You can simulate this with any .NET app (from 3.1 to 6.0) where you simply throw an exception in the Main method.

**Comment by bijeaylimbu** [comment]
https://github.com/dotnet/core/issues/7112#issuecomment-1033292899   thanks your solutions works

**Comment by omajid** [comment]
We are trying to fix our docs to warn people about snap being mostly broken for dotnet tools: https://github.com/dotnet/docs/pull/31045

**Comment by Sidjay00** [comment]
> The solution is weird, I don't know why. I have installed `dotnet` through `snap` and it was working fine, no problem there. Suddenly on February when I tried to run some projects, it didn't work, it showed `watch : Exited with error code 139`. I tried two possible solutions: the first one, I downloaded the binaries from [here](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) and I used the executable (for the versions 5 and 6). Same error. The second solution was to install from apt (I am using Ubuntu) using this [guide](https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#2004-). **It worked!**. Again, I don't know why, but I can run any project using `dotnet run` or `dotnet watch run`. Although this does not answer the question: why did that kind of error show up?

Thank you so much! The 2nd solution did helped me.

**Comment by nandafirmans** [comment]
I'm getting the same issue on archlinux, I'm using dotnet 6.0 I had no clue why this happen 

```
dotnet watch ❌ Exited with error code 139
dotnet watch ⏳ Waiting for a file to change before restarting dotnet...```

**Comment by danroth27** [comment]
@nandafirmans I don't do much on Linux, but it doesn't look like archlinux is currently on the supported distro list for .NET 8: https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md.

@richlander Do you know if there are known issues running .NET on archlinux?

**Comment by diegodfsd** [comment]
Does anyone have updates regarding this issue? We are currently using .NET Core SDK 7 (mcr.microsoft.com/dotnet/sdk:7.0) on Amazon Elastic Container Service (ECS). For several weeks now, we have been encountering these errors where the container is abruptly terminated with exit code 139. Unfortunately, there is no additional information provided to help us comprehend the root cause of the error.

**Comment by arhue** [comment]
Facing the same issue as @diegodfsd above. Tried Ubuntu amd64 and Alpine amd64 image. Same issue. Nothing in logs. Seems more easily triggered on Ubuntu vs Alpine, which takes longer. Tried debugging it for weeks now.

**Comment by huzaynbolt** [comment]
@arhue / @diegodfsd Do you have fix to this issue yet? It's being a bottle neck for one of my service with similar setup on ECS

**Comment by diegodfsd** [comment]
Hi @huzaynbolt we solved our problem. In our case, the cause behind the abruptly terminated ECS Tasks was the usage of the IronPdf library that uses chromium, but after we made the library upgrade the issue was fixed.

**Comment by huzaynbolt** [comment]
How do your debug on ECS then? That's my struggle at the moment. Since ECS terminated the task upon the error and I can't retrieve the logs after it stopped

**Comment by arhue** [comment]
@huzaynbolt Sorry about the delay, missed the notification. Yes, for us it was because application was running out of memory
